users/users.go

```diff
@@ -81,3 +81,40 @@ func SignInUser(w http.ResponseWriter, r *http.Request) {
 	}
 	w.Write(decodedResult)
 }
+
+func UpdateUser(w http.ResponseWriter, r *http.Request) {
+	input := mux.Vars(r)
+	Id := input["iD"]
+	collection := utils.GetCollection(utils.Users)
+	findResult := collection.FindOne(context.TODO(), Id)
+	if findResult.Err() != nil {
+		http.Error(w, "Unable to find the User data", http.StatusBadRequest)
+		return
+	}
+	var mongoUser models.User
+	err := findResult.Decode(&mongoUser)
+	if err != nil {
+		http.Error(w, "error occured during internal process", http.StatusInternalServerError)
+		return
+	}
+	var user models.User
+	err = json.NewDecoder(r.Body).Decode(&user)
+	if err != nil {
+		http.Error(w, "error occured during getting update attributes", http.StatusBadRequest)
+		return
+	}
+	switch user {
+	case user.Address != "":
+		mongoUser.Address = user.Address
+	case user.Password != "":
+		mongoUser.Password = user.Password
```

Comment:
Password update should be a different api, and it should be designed in future, [create this issue](https://github.com/abhishekhsheoran/library-management/issues/23), please remove password update from here and also restrict that it should not be updated from here with proper test cases.